### PR TITLE
fix(sdk): improvement of build-in process to delete nodes

### PR DESCRIPTION
**Pull request by xuelianhan007**
## Description

Added improvement of build-in process to delete nodes  with a null value in inventory response
## Related Issue

https://github.com/mycloudnexus/kraken/issues/34
## How This Has Been Tested
Integration test in localhost

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] CI/CD or documentation update (changes to CI/CD pipeline or documentation)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Review by kuangxiang20240501 — APPROVED** [review]
LGTM

File: kraken-app/kraken-app-hub/src/main/resources/mock/product.yaml [hunk 1]
```diff
@@ -55,7 +61,20 @@ spec:
     #- classpath:/mock/api-targets-mappers/api-target-mapper.product.offering.eline.add.yaml
     #- classpath:/mock/api-targets-mappers/api-target-mapper.product.offering.uni.read.yaml
     #- classpath:/mock/api-targets-mappers/api-target-mapper.product.offering.eline.read.yaml
-    - classpath:/mock/mapping-matrix/mapping.matrix.order.yaml
-    - classpath:/mock/mapping-matrix/mapping.matrix.quote.yaml
-    - classpath:/mock/mapping-matrix/mapping.matrix.address.validation.yaml
-
+    #- classpath:/mock/api-targets-mappers/api-target-mapper.inventory.eline.list.yaml
```
Comment (zhiyaoPCCW) [inline kraken-app/kraken-app-hub/src/main/resources/mock/product.yaml:64]: Please consider to remove commented out code

File: kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.order.notification.state.change.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+kind: kraken.component.api-target
+apiVersion: v1
+metadata:
+  key: mef.sonata.api-target.order.notification.state.change
```
Comment (zhiyaoPCCW) [inline kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.order.notification.state.change.yaml:5]: Please remove this file if it is useless

File: kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.poq.notification.state.change.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+kind: kraken.component.api-target
+apiVersion: v1
+metadata:
+  key: mef.sonata.api-target.poq.notification.state.change
```
Comment (zhiyaoPCCW) [inline kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.poq.notification.state.change.yaml:5]: same with above

File: kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.product.offering.eline.add.yaml [hunk 4]
```diff
@@ -5,6 +5,7 @@ metadata:
   key: mef.sonata.api-target.poq.eline.add
   name: Product Offering Qualification (POQ) API
   mapperKey: mef.sonata.api-target-mapper.poq.eline.add
+  version: 1
 spec:
   inputs:
```
Comment (zhiyaoPCCW) [inline kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.product.offering.eline.add.yaml:10]: same with above

File: kraken-app/kraken-app-hub/src/test/resources/mock/product.yaml [hunk 5]
```diff
@@ -55,6 +61,20 @@ spec:
     #- classpath:/mock/api-targets-mappers/api-target-mapper.product.offering.eline.add.yaml
     #- classpath:/mock/api-targets-mappers/api-target-mapper.product.offering.uni.read.yaml
     #- classpath:/mock/api-targets-mappers/api-target-mapper.product.offering.eline.read.yaml
-    #- classpath:/mock/mapping-matrix/mapping.matrix.order.yaml
-    #- classpath:/mock/mapping-matrix/mapping.matrix.quote.yaml
-    #- classpath:/mock/mapping-matrix/mapping.matrix.address.validation.yaml
\ No newline at end of file
+    #- classpath:/mock/api-targets-mappers/api-target-mapper.inventory.eline.list.yaml
```
Comment (zhiyaoPCCW) [inline kraken-app/kraken-app-hub/src/test/resources/mock/product.yaml:64]: Consider to remove commented out code.

**Review by zhiyaoPCCW — CHANGES_REQUESTED** [review]

File: kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.order.notification.state.change.yaml (hunk 2 above)
Comment (xuelianhan007) [reply kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.order.notification.state.change.yaml]: Thanks. This file has been deleted.

File: kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.poq.notification.state.change.yaml (hunk 3 above)
Comment (xuelianhan007) [reply kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.poq.notification.state.change.yaml]: Thanks. This file has been deleted.

File: kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.product.offering.eline.add.yaml (hunk 4 above)
Comment (xuelianhan007) [reply kraken-app/kraken-app-hub/src/test/resources/mock/api-targets/api-target.product.offering.eline.add.yaml]: Thanks. This file has been deleted.

File: kraken-app/kraken-app-hub/src/main/resources/mock/product.yaml (hunk 1 above)
Comment (xuelianhan007) [reply kraken-app/kraken-app-hub/src/main/resources/mock/product.yaml:64]: Inventory is in use now. So this line will be reserved for fast testing in future.

**Review by zhiyaoPCCW — APPROVED** [review]
LGTM

**Pull request merged**